include/multipass/availability_zone.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_AVAILABILITY_ZONE_H
+#define MULTIPASS_AVAILABILITY_ZONE_H
+
+#include "disabled_copy_move.h"
+#include "virtual_machine.h"
+
+#include <memory>
+
+namespace multipass
+{
+class AvailabilityZone : DisabledCopyMove
```

Comment:
Not a review, I just noticed this: private inheritance would be preferable in this case (and in the manager).

Reply:
https://www.boost.org/doc/libs/1_87_0/libs/core/doc/html/core/noncopyable.html

Reply:
Isn't `private` the default in this case?

Reply:
surprisingly, [nope](https://gcc.godbolt.org/z/nbPGEc4r8).

Reply:
@xmkg but we're dealing with classes here, not with structs https://gcc.godbolt.org/z/o5Mdbnsxs